src/platform.ts

```diff
@@ -20,96 +95,126 @@ export enum Platform {
     Ubuntu16
 }
 
-function convertOSReleaseTextToMap(text : string) : Map<string, string> {
-    let ret: Map<string, string> = new Map<string, string>();
-    const lines: string[] = text.split('\n');
-
-    for (let line of lines) {
-        line = line.trim();
-
-        let equalsIndex = line.indexOf('=');
-        if (equalsIndex >= 0) {
-            let key = line.substring(0, equalsIndex);
-            let value = line.substring(equalsIndex + 1);
+export class PlatformInformation {
+    constructor(
+        public operatingSystem: OperatingSystem,
+        public architecture: string,
+        public distribution: LinuxDistribution) { }
+
+    public static GetCurrent(): Promise<PlatformInformation> {
+        let operatingSystem: OperatingSystem;
+        let architecturePromise: Promise<string>;
+        let distributionPromise: Promise<LinuxDistribution>;
+
+        switch (os.platform()) {
+            case 'win32':
+                operatingSystem = OperatingSystem.Windows;
+                architecturePromise = PlatformInformation.GetWindowsArchitecture();
+                distributionPromise = Promise.resolve(null);
+                break;
 
-            // Strip double quotes if necessary
-            if (value.length > 1 && value.startsWith('"') && value.endsWith('"')) {
-                value = value.substring(1, value.length - 1);
-            }
+            case 'darwin':
+                operatingSystem = OperatingSystem.MacOS;
+                architecturePromise = PlatformInformation.GetUnixArchitecture();
+                distributionPromise = Promise.resolve(null);
+                break;
 
-            ret.set(key, value);
+            case 'linux':
+                operatingSystem = OperatingSystem.Linux; 
+                architecturePromise = PlatformInformation.GetUnixArchitecture();
+                distributionPromise = LinuxDistribution.GetCurrent();
+                break;
         }
```

Comment:
Yes, we should probably throw. The only other possible values are `aix`, `freebsd`, `openbsd` or `sunos` , which VS Code doesn't really support either.